### PR TITLE
Fix oracle kerberos testing on Java 23

### PR DESCRIPTION
**Pull request by KyleAure**
Prepare to update all tests to use Oracle JDBC 23 driver (not to be confused with Java 23 JDK)

### CODE REVIEW

File: dev/build.example.testcontainers_fat/build.gradle [hunk 1]
```diff
@@ -15,9 +15,22 @@
  // This bundle is required both at compiletime and runtime.
  addRequiredLibraries.dependsOn copyTestContainers
  
+ //TODO temporary while we switch from ojdbc8_g to ojdbc8
```
Comment (gjwatts) [inline dev/build.example.testcontainers_fat/build.gradle:18]: If the code that references the new Oracle JDBC driver is all introduced at the same time, why is a temporary solution needed?  There has to be some lag the temporary solution covers., but I don't understand why it is needed.

File: dev/build.example.testcontainers_fat/build.gradle [hunk 2]
```diff
@@ -15,9 +15,22 @@
...
+ task copyJdbcDriversOracleRename(type: Copy) {
+  mustRunAfter jar
+  from configurations.jdbcDrivers
+  into new File(autoFvtDir, 'publish/shared/resources/jdbc')
+  rename 'jcc.*.jar', 'jcc.jar'
+  rename 'derby-.*.jar', 'derby.jar'
+  rename 'derbyclient-.*.jar', 'derbyclient.jar'
+  rename 'ojdbc8_g.*.jar', 'ojdbc8.jar'
```
Comment (gjwatts) [inline dev/build.example.testcontainers_fat/build.gradle:26]: Why is copyJdbcDriversOracleRename only needed for ojdbc8 and not ojdbc11 as well?

File: dev/fattest.simplicity/src/componenttest/topology/database/container/DatabaseContainerType.java [hunk 3]
```diff
@@ -37,25 +38,27 @@
...
 public enum DatabaseContainerType {
-    DB2("jcc.jar", Db2Container.class.getCanonicalName(), Properties_db2_jcc.class, //
+    DB2("jcc.jar", "", Db2Container.class.getCanonicalName(), Properties_db2_jcc.class, //
         DockerImageName.parse("kyleaure/db2:1.0").asCompatibleSubstituteFor("ibmcom/db2")),
-    Derby("derby.jar", DerbyNoopContainer.class.getCanonicalName(), Properties_derby_embedded.class, DockerImageName.parse("")),
-    DerbyClient("derbyclient.jar", DerbyClientContainer.class.getCanonicalName(), Properties_derby_client.class, DockerImageName.parse("")),
-    Oracle("ojdbc8_g.jar", OracleContainer.class.getCanonicalName(), Properties_oracle.class, //
+    Derby("derby.jar", "", DerbyNoopContainer.class.getCanonicalName(), Properties_derby_embedded.class, DockerImageName.parse("")),
+    DerbyClient("derbyclient.jar", "", DerbyClientContainer.class.getCanonicalName(), Properties_derby_client.class, DockerImageName.parse("")),
+    Oracle("ojdbc8_g.jar", "ojdbc8.jar", OracleContainer.class.getCanonicalName(), Properties_oracle.class, //
```
Comment (gjwatts) [inline dev/fattest.simplicity/src/componenttest/topology/database/container/DatabaseContainerType.java:45]: Why do you have the old `ojdbc8_g.jar` driver listed as the primary one and the new one as the alternate one?  Are you going to switch the places of the `ojdbc8_g.jar` and `ojdbc8.jar` in the method arguments when you remove the temporary solution?

**Comment by gjwatts** [comment]
Spoke to Kyle about this and he explained the necessity of the temporary changes.

**Review by gjwatts — APPROVED** [review]

File: dev/build.example.testcontainers_fat/build.gradle (hunk 2 above)
Comment (KyleAure) [reply dev/build.example.testcontainers_fat/build.gradle:26]: This was added just to test the temporary code added to fattest.simplicity.

File: dev/fattest.simplicity/src/componenttest/topology/database/container/DatabaseContainerType.java (hunk 3 above)
Comment (KyleAure) [reply dev/fattest.simplicity/src/componenttest/topology/database/container/DatabaseContainerType.java:45]: This temporary code is so we can make updates to our internal WebSphere Liberty project.

**Comment by KyleAure** [comment]
#libby - check headers
#build - personal build

**Comment by KyleAure** [comment]
Pipeline request for Java 23: https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=bcdb3c93-fb8e-424d-9997-7f5885fb435b&tab=Steps

Target locations of links might be accessible only to IBM employees.

**Comment by KyleAure** [comment]
Build status:
- Pipeline (default): failed with one known defect, failed with one intermittent defect
- Pipeline (java 23): Passed
- RTC build: failed with one known defect

**Pull request merged**